lte/gateway/c/core/oai/tasks/nas/esm/esm_ebr.c

```diff
@@ -651,16 +651,16 @@ bool esm_ebr_is_not_in_use(emm_context_t* emm_context, ebi_t ebi) {
  **      Others:    None                                       **
  **                                                                        **
  ***************************************************************************/
-static int esm_ebr_get_available_entry(emm_context_t* emm_context) {
+static status_or_int_t esm_ebr_get_available_entry(emm_context_t* emm_context) {
   ue_mm_context_t* ue_mm_context =
       PARENT_STRUCT(emm_context, struct ue_mm_context_s, emm_context);
   int i;
 
   for (i = 0; i < BEARERS_PER_UE; i++) {
     if (ue_mm_context->bearer_contexts[i] == NULL) {
-      return i;
+      return (status_or_int_t){RETURNok, i};
     }
   }
 
-  return -1;
+  return (status_or_int_t){RETURNerror, 0};
```

Comment:
I'm confused by the usage of this, is this supposed to emulate a tuple return of status and a value? Or this means that it could return either one of them? 

If this is the case of the latter, RETURNerror should have a different value of 0 right?

Reply:
This returns a status + value. If `RETURNerror` is included as the status, then the value is just arbitrary, as shown here. Ideally `status_or_int` would have been a union type, but there's no way to check then if it's a status that's returned, or an int.

Reply:
I've replaced this with a macro and more comments to make this clearer

Reply:
`RETURN_INT_ERROR`